### ISSUE TITLE
runc sometimes fails with "no such file or directory"

### ISSUE THREAD

**Issue by prezha** [kubernetes/minikube#17976]
### What Happened?

we see these in some kvm+containerd tests:

[example 1](https://storage.googleapis.com/minikube-builds/logs/17965/32726/KVM_Linux_containerd.html#fail_TestAddons%2fparallel%2fRegistry):
```
	I0116 02:30:24.180449  339661 ssh_runner.go:195] Run: sudo runc --root /run/containerd/runc/k8s.io list -f json
...
	stderr:
	time="2024-01-16T02:30:24Z" level=error msg="stat /run/containerd/runc/k8s.io/4744173521ad7687144254eea07ff8aff4223eee041b7cec0b7bda79fc00a1d0: no such file or directory"
````

[example 2](https://storage.googleapis.com/minikube-builds/logs/17967/32725/KVM_Linux_containerd.html#fail_TestAddons%2fparallel%2fHeadlamp):
```
	I0116 01:58:56.995758  567664 ssh_runner.go:195] Run: sudo runc --root /run/containerd/runc/k8s.io list -f json
...
	stderr:
	time="2024-01-16T01:58:57Z" level=error msg="stat /run/containerd/runc/k8s.io/7010e3c493ae1d4af36adeaef2828f0f9da1b15cdf9f78f705d7c137e0446ba3: no such file or directory"
```

we use runc v1.1.10, and once released, some future version could possibly solve this issue:

(our) runc v1.1.10: https://github.com/opencontainers/runc/blob/18a0cb0f32bcac2ecc9a10f327d282759c144dab/list.go#L131-L134
```
			st, err := os.Stat(filepath.Join(absRoot, item.Name()))
			if err != nil {
				fatal(err)
			}
```

(latest) runc v1.1.11 - unchanged: https://github.com/opencontainers/runc/blob/4bccb38cc9cf198d52bebf2b3a90cd14e7af8c06/list.go#L131-L134
```
			st, err := os.Stat(filepath.Join(absRoot, item.Name()))
			if err != nil {
				fatal(err)
			}
```

(current) runc HEAD: https://github.com/opencontainers/runc/blob/0c5a73535503216a8c15a86aa9022d3b0d995994/list.go#L129-L136
```
		st, err := item.Info()
		if err != nil {
			if errors.Is(err, os.ErrNotExist) {
				// Possible race with runc delete.
				continue
			}
			return nil, err
		}
```

so, it's due to a `Possible race with runc delete`

interestingly, this change was merged in pr [3379](https://github.com/opencontainers/runc/pull/3349) by @kolyshkin  on 2022-01-27, but it looks like it was never released (see the current "runc v1.1.11" above, released on 2024-01-02)

we could simply retry until this gets fixed/released upstream, and then we update to that runc version

**Comment by kolyshkin** [comment]
runc 1.1 backport: https://github.com/opencontainers/runc/pull/4231 (will be released in 1.1.13 if/when we do another 1.1.x release)

**Comment by prezha** [comment]
closing this issue as it was fixed upstream and we've bumped runc to pick that up
details [here](https://github.com/kubernetes/minikube/pull/17977#issuecomment-2195790635)
